lace_grid.py

```diff
@@ -13,131 +13,152 @@
+                }
+        # create attributes from style and define path
+        attribs = {'style':simplestyle.formatStyle(style), 
+                   'cx':str(x),
+                   'cy':str(y),
+                   'r':str(r)
+                  }
+        # insert path object into current layer
+        inkex.etree.SubElement(parent, inkex.addNS('circle', 'svg'), attribs)
+
+    def draw_grid_pin(self, x, y, parent):
+        " Draw a single grid pin "
+        pin_radius = self.options.pindia/2
+        fill = self.options.pincolor
+        self.circle(x, y, pin_radius, fill, '#000000',  '0.1mm', parent)
+
+    def draw_grid(self, width, height, spacing, theta, parent):
+        " Draw a grid of pins "
+        hgrid = spacing*sin(theta);
+        vgrid = spacing*cos(theta)
+        rows = int(height / vgrid) + 1
+        cols = int(width  / hgrid)
+        y = 0.0
+        for r in range(rows):
+            x = 0.0
+            if (r % 2 == 1):
+                x += hgrid
+            for c in range(cols/2):
+                self.draw_grid_pin(x, y, parent)
+                x += 2.0*hgrid;
+            y += vgrid;
     
-	def effect(self):
-		"""
-		Effect behaviour.
-		Overrides base class' method and draws something.
-		"""
-		# Convert input from mm to pixels, assuming 90 dpi
-		conversion = 90.0 / 25.4
-		self.options.width *= conversion
-		self.options.height *= conversion
-		self.options.size *= conversion / 2.0 # also convert from diameter to radius
-		
-		# Users expect spacing to be the vertical distance between footside pins (vertical distance between every other row) 
-		# but in the script we use it as as diagonal distance between grid points
-		# therefore convert spacing based on the angle chosen
-		theta = radians(self.options.angle)
-		self.options.spacing *= conversion/(2.0*cos(theta))
-		
-		# Convert color from long integer to hexidecimal string
-		self.options.color = self.getColorString(self.options.color)
-		
-		# Draw a grid of dots based on user inputs
-		self.draw()
+    def effect(self):
+        """
+        Effect behaviour.
+        Overrides base class' method and draws something.
+        """
+        #Convert input from mm or whatever user uses
+        conversion = self.getUnittouu("1" + self.options.units)
+        width = self.options.width * conversion
+        height = self.options.height * conversion
+        # sort out color
+        self.options.pincolor = self.getColorString(self.options.pincolor)
+        self.options.pindia = self.options.pindia * conversion
+        # users expect spacing to be the vertical distance between footside pins (vertical distance between every other row) 
+        # but in the script we use it as as diagonal distance between grid points
+        # therefore convert spacing based on the angle chosen
+        theta = radians(self.options.angle)
+        spacing = self.options.spacing * conversion/(2.0*cos(theta))
+
+        # Top level Group

```

Comment:
If it makes more sense to sub group these in lines or by some other feature - please let me know.
For now I have added a single group holding all the dots.
IMHO This makes it easier to pick up and move.